interfaces/snap_app_set.go

```diff
@@ -104,38 +104,36 @@ func (a *SnapAppSet) SecurityTagsForSlot(slot *snap.SlotInfo) ([]string, error)
 	return tags, nil
 }
 
-// PlugLabelExpression returns the label expression for the given plug. It is
+// plugLabelExpression returns the label expression for the given plug. It is
 // constructed from the apps and hooks that are associated with the plug.
-func (a *SnapAppSet) PlugLabelExpression(plug *ConnectedPlug) string {
+func (a *SnapAppSet) plugLabelExpression(plug *ConnectedPlug) string {
 	// TODO: this is a hack that will not continue to work once component hooks
 	// are introduced. the methods on SnapAppSet should only be called on
 	// slots/hooks that originated from the snap that the SnapAppSet was derived
 	// from.
-	info := a.info
 	if a.info.InstanceName() != plug.plugInfo.Snap.InstanceName() {
```

Comment:
shouldn't this use plug.appSet now?  and the other check go to the constructors of ConnectedPlug?

Reply:
We could actually get rid of the `snap.Info` param to `NewConnectedPlug/Slot`, since the app set already has a pointer to the snap info.

Reply:
Actually I can't read, that wouldn't work.

Reply:
Would you want to panic in that constructor, or return an error? It is used quite a bit in tests, so it'd be a largish diff if we return an error. Not too much compared to the rest of the refactor though.

Reply:
I think panic is ok